### PR TITLE
MAINT: Update dependencies and revitalize the CI pipeline

### PR DESCRIPTION
**Pull request by florin-pop**
This PR fixes https://github.com/mne-tools/mne-nirs/issues/506 by introducing the following changes:
 - have pip install packages directly from `git` instead of GitHub tarballs or zipballs
 - replace `footer_items` (which is deprecated) with `footer_start` in `doc/conf.py`
 - fixes a typo
 - install `libxcb-cursor0` in addition to the rest of libxcb dependencies

These fixes were mostly adaptations of changes introduced by @larsoner in https://github.com/mne-tools/mne-python/

---
Contributed with ❤️ by AE Studio

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -13,7 +13,7 @@ jobs:
       docker:
         # Use 18.04 rather than 20.04 because MESA 20.0.8 on 18.04 has working
         # transparency but 21.0.3 on 20.04 does not!
-        - image: cimg/base:stable-18.04
+        - image: cimg/base:2023.03-18.04
```
Comment (florin-pop) [inline .circleci/config.yml:16]: I'm not sure that this change is needed, I'll revert it and try again.

File: .github/workflows/linux_pip.yml [hunk 2]
```diff
@@ -27,6 +27,7 @@ jobs:
       OPENBLAS_NUM_THREADS: '1'
       PYTHONUNBUFFERED: '1'
       PYTHON_VERSION: '3.10'
+      QT_DEBUG_PLUGINS: '1'
     strategy:
```
Comment (florin-pop) [inline .github/workflows/linux_pip.yml:31]: This will be reverted as well before merge.

File: requirements.txt [hunk 3]
```diff
@@ -33,7 +33,7 @@ xlrd
 imageio>=2.6.1
 imageio-ffmpeg>=0.4.1
 traitlets
-pyvista>=0.32,!=0.35.2,!=0.38.0,!=0.38.1,!=0.38.2,!=0.38.3
+pyvista>=0.32,!=0.35.2,!=0.38.0,!=0.38.1,!=0.38.2,!=0.38.3,!=0.38.4,!=0.38.5
```
Comment (florin-pop) [inline requirements.txt:36]: I'm not happy with this change. What do you think about pinning pyvista to `>=0.32,<0.38`?

Comment (larsoner) [reply requirements.txt:36]: Agreed it's ugly, but IIRC the bug is/should be fixed in their `main` so it should work with the next release 🤷

File: .circleci/config.yml (hunk 1 above)
Comment (larsoner) [reply .circleci/config.yml:16]: This comment is very old and IIRC 22.04 mesa might actually be okay. Here I'd do whatever we do in MNE-Python's CircleCI config as things are tested/updated much more often there...

Comment (larsoner) [reply .circleci/config.yml:16]: (you'll need to make sure you have the MNE_3D_OPTION stuff equivalent between the two... IIRC we discovered the transparency bug was due to full-screen antialiasing being enabled or something)

File: requirements.txt (hunk 3 above)
Comment (rob-luke) [reply requirements.txt:36]: Let's stick with ugly then

**Comment by rob-luke** [comment]
Thanks @florin-pop it looks good to me. I also looked through the rendered docs and they were correct (including the 3d renders).

Could you please revert the change you wanted to revert and then ping Eric to merge.

File: .github/workflows/linux_pip.yml (hunk 2 above)
Comment (rob-luke) [reply .github/workflows/linux_pip.yml:31]: Please leave in draft PR merge while there are things you intend to revert so we don't accidentally merge things that are intended for debugging.

**Review by rob-luke — APPROVED** [review]

File: .github/workflows/linux_pip.yml (hunk 2 above)
Comment (rob-luke) [reply .github/workflows/linux_pip.yml]: I realised I have access to revert the change ;) Done! THanks Florin

**Comment by rob-luke** [comment]
> Could you please revert the change you wanted to revert and then ping Eric to merge.

I remembered I can revert the change myself ;) So we don't need to bug Eric now and I'll merge when all green

**Comment by rob-luke** [comment]
This is absolutely super! Thanks @florin-pop

**Pull request merged**